bench/benches/packets.rs

```diff
@@ -9,39 +9,124 @@ use proptest::prelude::*;
 
 const BATCH_SIZE: usize = 500;
 
-fn parse_udp_packet(mbuf: Mbuf) -> Udp<Ipv4> {
+fn parse_packet(mbuf: Mbuf) -> Udp<Ipv4> {
     let ethernet = mbuf.parse::<Ethernet>().unwrap();
     let ipv4 = ethernet.parse::<Ipv4>().unwrap();
     ipv4.parse::<Udp<Ipv4>>().unwrap()
 }
 
-fn deparse_udp_packet(udp: Udp<Ipv4>) -> Mbuf {
+fn deparse_packet(udp: Udp<Ipv4>) -> Mbuf {
     let d_ipv4 = udp.deparse();
     let d_eth = d_ipv4.deparse();
     d_eth.deparse()
 }
 
-#[nb2::bench(mempool_capacity = 512)]
-fn parse_benchmark(c: &mut Criterion) {
-    c.bench_function("packets::parse_udp_packets", move |b| {
+fn peek_through_packet(mbuf: Mbuf) -> Mbuf {
+    let ethernet = mbuf.peek::<Ethernet>().unwrap();
+    let ipv4 = ethernet.peek::<Ipv4>().unwrap();
+    ipv4.peek::<Udp<Ipv4>>().unwrap();
+    mbuf
+}
+
+fn peek_into_packet(ipv4: Ipv4) -> Ipv4 {
+    ipv4.peek::<Udp<Ipv4>>().unwrap();
+    ipv4
+}
+
+fn reset_packet(udp: Udp<Ipv4>) -> Mbuf {
+    udp.reset()
+}
+
+fn push_into_mbuf(mbuf: Mbuf) -> Udp<Ipv4> {
+    let ethernet = mbuf.push::<Ethernet>().unwrap();
+    let ipv4 = ethernet.push::<Ipv4>().unwrap();
+    ipv4.push::<Udp<Ipv4>>().unwrap()
+}
+
+fn push_into_packet(ipv4: Ipv4) -> Udp<Ipv4> {
+    ipv4.push::<Udp<Ipv4>>().unwrap()
+}
+
+fn remove_from_packet(udp: Udp<Ipv4>) -> Ipv4 {
+    udp.remove().unwrap()
+}
+
+#[nb2::bench(mempool_capacity = 511)]
+fn parse_benchmarks(c: &mut Criterion) {
+    c.bench_function("packets::parse_packets", |b| {
         let s = v4_udp();
-        b.iter_batches(BATCH_SIZE, s, parse_udp_packet)
+        b.prop_iter_batched(BATCH_SIZE, s, parse_packet)
     });
 
-    c.bench_function("packets::deparse_udp_packets", move |b| {
+    c.bench_function("packets::deparse_packets", |b| {
```

Comment:
if we want to compare results of two bench functions, we should maybe use `benchmark_group`. although in this case, parse and deparse can't really compare. maybe should be two different benches.

https://docs.rs/criterion/0.3.0/criterion/struct.Criterion.html#method.benchmark_group